pipeline/test/test_extract.py

```diff
@@ -0,0 +1,96 @@
+import os
+import json
+from pathlib import Path
+import pytest
+from extract.document import Document, TextBlock, BlockCoordinates, SEPARATOR
+from extract.extract import DocumentEmbeddedTextExtractor
+
+
+@pytest.fixture
+def test_pdf_path():
+    return Path(__file__).parent / "data/cclw-1318-d7f66920a18e4ddf94c83cf21fa2bcfa.pdf"
+
+
+@pytest.fixture
+def document():
+    return Document(
+        text_blocks=[
+            TextBlock(
+                "paragraph 1",
+                "p1_b1",
+                "p1",
+                coords=[
+                    BlockCoordinates(1, 1),
+                    BlockCoordinates(1, 2),
+                    BlockCoordinates(2, 2),
+                    BlockCoordinates(2, 1),
+                ],
+            )
+        ],
+        filename="test_document.pdf",
+        dimensions=BlockCoordinates(4, 4),
+    )
+
+
+def test_document_save_json(document, tmpdir):
+    output_path = os.path.join(tmpdir, "test_document.json")
+
+    document.save_json(output_path)
+
+    assert os.path.exists(output_path)
+    assert len(tmpdir.listdir()) == 1
+
+    with open(output_path, "r") as f:
+        doc_data = json.load(f)
+
+    assert doc_data == {
+        "filename": "test_document.pdf",
+        "dimensions": {"x": 4, "y": 4},
+        "text_blocks": [
+            {
+                "text": "paragraph 1",
+                "text_block_id": "p1_b1",
+                "page_id": "p1",
+                "coords": [
+                    {"x": 1, "y": 1},
+                    {"x": 1, "y": 2},
+                    {"x": 2, "y": 2},
+                    {"x": 2, "y": 1},
+                ],
+            }
+        ],
+    }
+
+
+def test_document_to_string(document):
+
+    assert document.to_string() == SEPARATOR + "paragraph 1" + "\n"
+
+
+def test_document_save_text(document, tmpdir):
+    output_path = os.path.join(tmpdir, "test_document.txt")
```

Comment:
I know it's only `/tmp`, but might still be worth cleaning up when the test finishes?

Reply:
It seems like pytest only keeps [max 3 temporary folders](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-default-base-temporary-directory), and I'm thinking it might be useful to keep these around in case a test fails?

What's the advantage of cleaning up after the test finishes/is it best practice?

Reply:
Ah, please ignore me. Just realised it's the pytest `tmpdir` fixture.